bigtable/admin.go

```diff
@@ -2150,9 +2150,60 @@ func (ac *AdminClient) RestoreTableFrom(ctx context.Context, sourceInstance, tab
 	return longrunning.InternalNewOperation(ac.lroClient, op).Wait(ctx, &resp)
 }
 
+type backupOptions struct {
+	backupType        *BackupType
+	hotToStandardTime *time.Time
+}
+
+// BackupOption can be used to specify parameters for backup operations.
+type BackupOption interface {
+	apply(o *backupOptions)
+}
+
+// setHotToStandardTime can be used to set backup type to [BackupTypeHot] and the
+// time at which the hot backup will
+// be converted to a standard backup.
+// Once the `hot_to_standard_time` has passed, Cloud Bigtable will convert the
+// hot backup to a standard backup. This value must be greater than the backup
+// creation time by at least 24 hours.
+type hotBackupOption struct {
+	htsTime *time.Time
+}
+
+func (hbo hotBackupOption) apply(o *backupOptions) {
+	o.hotToStandardTime = hbo.htsTime
+	btHot := BackupTypeHot
+	o.backupType = &btHot
+}
+
+// HotToStandardBackup option can be used to create backup with
+// type [BackupTypeHot] and specify time at which the hot backup will be
+// converted to a standard backup. Once the `hot_to_standard_time` has passed,
+// Cloud Bigtable will convert the hot backup to a standard backup.
+// This value must be greater than the backup creation time by:
+// - At least 24 hours
+func HotToStandardBackup(hotToStandardTime time.Time) BackupOption {
+	return hotBackupOption{htsTime: &hotToStandardTime}
+}
+
+// HotBackup option can be used to create backup
+// with type [BackupTypeHot]
+func HotBackup() BackupOption {
+	return hotBackupOption{}
+}
+
 // CreateBackup creates a new backup in the specified cluster from the
 // specified source table with the user-provided expire time.
 func (ac *AdminClient) CreateBackup(ctx context.Context, table, cluster, backup string, expireTime time.Time) error {
+	return ac.createBackup(ctx, table, cluster, backup, expireTime, nil)
+}
+
+// CreateBackupWithOptions is similar to CreateBackup but lets the user specify additional options.
+func (ac *AdminClient) CreateBackupWithOptions(ctx context.Context, table, cluster, backup string, expireTime time.Time, opts ...BackupOption) error {
+	return ac.createBackup(ctx, table, cluster, backup, expireTime, opts...)
+}
+
+func (ac *AdminClient) createBackup(ctx context.Context, table, cluster, backup string, expireTime time.Time, opts ...BackupOption) error {
```

Comment:
if we go with adding `expireTime` to `backupOptions`, then we can have the original `CreateBackup` (that takes `expireTime` as an arg) create its own `backupOptions` and pass it to this function

and then we won't need `expireTime` as an arg to this function